### PR TITLE
Protect from null pointer exeption and silent failure

### LINKED ISSUE #12304: null is not an object (evaluating 'this.Y.removeAll')
Looks like dispose is somehow called more than once in `visibility-model.js`.

/cc @zhouyx

### PR DESCRIPTION
**Pull request by jonkeller**
Fixes #12304

### CODE REVIEW

File: extensions/amp-analytics/0.1/visibility-model.js [hunk 1]
```diff
@@ -80,6 +80,11 @@ export class VisibilityModel {
               `Cannot repeat with interval less than ${MIN_REPEAT_INTERVAL}, ` +
               ' repeat set to false');
         }
+      } else {
```
Comment (lannka) [inline extensions/amp-analytics/0.1/visibility-model.js]: revert this?

Comment (jonkeller) [inline extensions/amp-analytics/0.1/visibility-model.js]: FYI going to kill the whole if/else on line 65 next, and make repeat only be allowed to be a boolean, not a number. But probably won't get to that before the canary that's in 2 hours.

Comment (lannka) [reply extensions/amp-analytics/0.1/visibility-model.js]: you might want to keep this PR simple for easy cherry-pick

Comment (jonkeller) [reply extensions/amp-analytics/0.1/visibility-model.js]: Ok, will do

File: extensions/amp-analytics/0.1/visibility-model.js [hunk 2]
```diff
@@ -80,7 +80,6 @@ export class VisibilityModel {
               `Cannot repeat with interval less than ${MIN_REPEAT_INTERVAL}, ` +
               ' repeat set to false');
         }
-      }
```
Comment (jonkeller) [inline extensions/amp-analytics/0.1/visibility-model.js]: Oops

Comment (jonkeller) [reply extensions/amp-analytics/0.1/visibility-model.js]: Fixed

File: extensions/amp-analytics/0.1/visibility-model.js (hunk 1 above)
Comment (jonkeller) [reply extensions/amp-analytics/0.1/visibility-model.js]: Done

Comment (jonkeller) [reply extensions/amp-analytics/0.1/visibility-model.js]: Done

File: extensions/amp-analytics/0.1/visibility-model.js [hunk 3]
```diff
@@ -239,8 +239,12 @@ export class VisibilityModel {
     });
     this.unsubscribe_.length = 0;
     this.eventResolver_ = null;
-    this.onTriggerObservable_.removeAll();
-    this.onTriggerObservable_ = null;
+    // TODO(jonkeller): Investigate why dispose() can be called twice,
+    // necessitating this "if"
+    if (this.onTriggerObservable_) {
+      this.onTriggerObservable_.removeAll();
```
Comment (calebcordry) [inline extensions/amp-analytics/0.1/visibility-model.js]: Should we also add warning here like in #12547 ?

**Comment by jonkeller** [comment]
Agreed. Just added to #12547.
https://github.com/ampproject/amphtml/pull/12547/files#diff-61c963752cc0a61e2ec5e1561dc0466cR217

**Comment by calebcordry** [comment]
I deleted the comment because I saw that you were one step ahead of me as soon as it was sent :)

**Pull request merged**